### ISSUE TITLE
Snap package does not install icon in Ubuntu 16.04

### ISSUE THREAD

**Issue by DivineOmega** [brave/browser-laptop#13155]
### Description

Installing Brave through snap on Ubuntu 16.04 does not result in an icon in the Unity dash.

### Steps to Reproduce

  1. Install brave via snap as per documented instructions: https://github.com/brave/browser-laptop/blob/master/docs/linuxInstall.md#snapcraft
  2. Open the dash with the Super key, type 'brave'.
  3. Note that no Brave icon appears.

**Actual result:**

Snap package installation does not result in a Brave icon.

**Expected result:**

Snap package should setup an icon for Brave.

**Reproduces how often:**

100%

### Brave Version

**about:brave info:**

Brave: 0.20.42 
V8: 6.4.388.41 
rev: 096c7cb3c75ebb518f72fc7d24bc2bbaedd50aed 
Muon: 4.8.2 
OS Release: 4.13.0-32-generic 
Update Channel: Release 
OS Architecture: x64 
OS Platform: Linux 
Node.js: 7.9.0 
Brave Sync: v1.4.2 
libchromiumcontent: 64.0.3282.140

**Reproducible on current live release:**

Yes

**Comment by Jacalz** [comment]
Will have a look at reproducing this issue, hang tight 🙂

**Comment by Jacalz** [comment]
I was successfully able to reproduce but I could not even start brave with "snap run brave"

**Comment by Garbee** [comment]
The problem is since it is the generic release. For example the [0.23.73 build](https://github.com/brave/browser-laptop/releases/download/v0.23.73/Brave.tar.bz2). Download and extract this, then `./brave` in the content directory, yields no icon in the system interfaces. The generic builds would need to be made to set the icon when the browser starts.

**Comment by Garbee** [comment]
Ok, I got some help from someone on a Flatpak build I'm working on. Brave spawns its window names as `brave`, all lower case. So if you set `StartupWMClass=brave` in the `.desktop` file in the snap, that will get the icon on the spawned windows.

I'm not sure who maintains the snap build and how to let them know. But that's the fix.

I figured this out by being pointed to the `StartupWMClass` option for desktop files. Then using `xwininfo -tree -root | grep Brave` and seeing how the windows were named.
